backend/functions/src/emailing/functions.ts

```diff
@@ -1,4 +1,36 @@
 import axios from 'axios'
+
+// firebase imports
+import admin from 'firebase-admin'
+import { db } from '../config'
+const courseRef = db.collection('courses')
+
+/** Updating Email Sent Timestap: 
+ * @param courseId is the string courseId usually in the form of a six-digit number such as 358546. 
+ * @param groups is the groups for the given course that emails were sent to. 
+      Currently type-checked as a string [] but may be a number [] depending on how the frontend obtains group number. 
+ * @result updates database to have email sent timestamp to current time. 
+ *  */
+export const updateEmailTimestamp = async (
+  courseId: string,
+  groups: string[]
+) => {
+  const time = admin.firestore.FieldValue.serverTimestamp()
+
+  // either groups = [0, 1, 2] or ["0", "1", "2"]
+  // must be string format -> parse here or when calling function
+  await Promise.all([
+    groups.forEach((group: string) => {
+      courseRef
+        .doc(courseId)
+        .collection('groups')
+        .doc(group)
+        .update({ shareMatchEmailTimestamp: time })
+    }),
+  ])
```

Comment:
Not sure if the Promises here work, I think it's something more like

```js
await Promise.all(
  groups.map((group: string) =>
    courseRef
      .doc(courseId)
      .collection('groups')
      .doc(group)
      .update({ shareMatchEmailTimestamp: time })
  ),
)
```

Note that I omitted the curly braces and return statement since the group numbers get mapped to Promises directly, which are then given to Promise.all